### PR TITLE
Work-around for https://github.com/couchbase/CouchbaseMock/issues/11.…

### PR DESCRIPTION
**Pull request by jimhooker2002**
…  This allows

clients to be setup without having to do the carrier port dance via get_mcports.

### CODE REVIEW

**Comment by jimhooker2002** [comment]
CLA signed in Gerrit.  Hopefully this can now be reviewed.

**Comment by mnunberg** [comment]
Make sure your Gerrit email matches the email you used in git when you did `git commit`. I don't see this PR popping up in Gerrit yet.

That being said, I'd accept this if this had a mandatory `bucket` argument. So it'd be `getCarrierPort("default")` for example.

File: src/main/java/org/couchbase/mock/CouchbaseMock.java [hunk 1]
```diff
@@ -261,6 +261,21 @@ public int getHttpPort() {
...
+     * Get the carrier port for a bucket.
+     * @param bucketName The bucket name to fetch the carrier port for.
+     * @return The carrier port.
+     */
+    public int getCarrierPort(String bucketName) {
+        Bucket bucket = buckets.get(bucketName);
+        if (null == bucket) {
+            // Buckets are created when the mock is started.  Calling getCarrierPort()
+            // before the mock has been started makes no sense.
+            throw new RuntimeException("Bucket does not exist.  Has the mock been started?");
```
Comment (mnunberg) [inline src/main/java/org/couchbase/mock/CouchbaseMock.java]: nit: extraneous space

Comment (jimhooker2002) [inline src/main/java/org/couchbase/mock/CouchbaseMock.java]: The nits have now been nuked.

**Comment by jimhooker2002** [comment]
OK, turns out I wasn't doing very good computering yesterday.  I signed up to Gerrit with the wrong email (as you correctly ascertained).  I've corrected that now and signed the CLA.  Still don't see the review showing up in my account there, but I guess it takes a bit-o-time for cb-sdk-robot to do what cb-sdk-robot does.

If it doesn't show up by end of day today, then I'll poke more buttons and throw more levers in the vain hope that something will work.  Basically the same strategy I use to solve a Rubik's Cube.

**Comment by jimhooker2002** [comment]
Hmm, Gerrit still no-likey, my review request doesn't appear to show up there for some reason.  I'm not sure what I can do.  Does it usually take this long for Gerrit to get-it (haha)?  Waiting a bit more.....

**Comment by mnunberg** [comment]
It might be easier for you to just interface directly with Gerrit. You might also need to squash your commits and do something like git push --force

**Comment by jimhooker2002** [comment]
git push --force attempted.  Waiting, waiting, waiting....

**Comment by jimhooker2002** [comment]
Still no luck.  I'm going to close this pull request and open an another to see if Gerrit likes that any more.

**Pull request closed without merging**